server/engine/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolPreprocessor.g4

```diff
@@ -10,7 +10,7 @@ parser grammar CobolPreprocessor;
 options {tokenVocab = CobolPreprocessorLexer;}
 
 startRule
-   : .*? ((includeStatement | copyStatement | replaceAreaStart | replaceOffStatement | titleDirective | enterDirective
+   : .*? ((includeStatement | copyStatement | replaceAreaStartOrOffStatement | titleDirective | enterDirective
    | controlDirective | linkageSection | plusplusIncludeStatement | procedureDivision | workingStorageSection)+ .*?)* EOF
```

Comment:
Did you measure how much this helps?
As far as I can tell, the source of slowdown is the combination of `procedureDivision` and `.*?)*`.

Reply:
Not precisely, about 10%. You are right about .*, my tests shows the same. I was planning to resolve it in other PR.